custom_model_runner/datarobot_drum/drum/typeschema_validation.py

```diff
@@ -374,25 +371,19 @@ def get_type_schema_yaml_validator():
     )
 
 
-def revalidate_typeschema(type_schema):
+def revalidate_typeschema(type_schema: YAML):
     """Perform validation on each dictionary in the both lists.  This is required due to limitations in strictyaml.  See
     the strictyaml documentation on revalidation for details.  This checks that the provided values
     are valid together while the initial validation only checks that the map is in the right general format."""
-    input_validation = {
-        d.FIELD: d.get_yaml_validator()
-        for d in [DataTypes, SparsityInput, NumColumns, InputContainsMissing]
-    }
-    output_validation = {
-        d.FIELD: d.get_yaml_validator()
-        for d in [DataTypes, SparsityOutput, NumColumns, OutputContainsMissing]
-    }
-    if "input_requirements" in type_schema:
-        for req in type_schema["input_requirements"]:
-            req.revalidate(input_validation[req.data["field"]])
-    if "output_requirements" in type_schema:
-        for req in type_schema["output_requirements"]:
-            print(req.data)
-            req.revalidate(output_validation[req.data["field"]])
+
+    for input_req in type_schema.get("input_requirements", []):
+        field = Fields.from_string(input_req.data["field"])
+        requirements = field.to_input_requirements()
+        input_req.revalidate(requirements)
```

Comment:
I hate the word revalidate a lot, but I understand why we have it

Reply:
yup.  also, that is `strict_yaml`'s revalidate.  what i _really_ hate is that revalidate **mutates** the req.